src/native/libs/System.Globalization.Native/pal_locale.m

```diff
@@ -29,6 +32,152 @@
 }
 
 #if defined(TARGET_MACCATALYST) || defined(TARGET_IOS) || defined(TARGET_TVOS)
+
+const char* NativeGetLocaleName(const char* localeName,
+                                         int32_t valueLength)
+{
+     NSString *locName = [NSString stringWithFormat:@"%s", localeName];
+     NSLocale *currentLocale = [[NSLocale alloc] initWithLocaleIdentifier:locName];
+     const char* value = [currentLocale.localeIdentifier UTF8String];
+     return strdup(value);
+}
+
+const char* NativeGetLocaleInfoString(const char* localeName,
+                                                LocaleStringData localeStringData,
+                                                int32_t valueLength,
+                                                const char* uiLocaleName)
+{
+    const char* value;
+    NSString *locName = [NSString stringWithFormat:@"%s", localeName];
+    NSLocale *currentLocale = [[NSLocale alloc] initWithLocaleIdentifier:locName];
+    NSNumberFormatter *numberFormatter = [[NSNumberFormatter alloc] init];
+    numberFormatter.locale = currentLocale;
+    NSDateFormatter* dateFormatter = [[NSDateFormatter alloc] init];
+    [dateFormatter setLocale:currentLocale];
+    NSLocale *gbLocale = [[NSLocale alloc] initWithLocaleIdentifier:@"en_GB"];
+
+
+    switch (localeStringData)
+    {
+        ///// <summary>localized name of locale, eg "German (Germany)" in UI language (corresponds to LOCALE_SLOCALIZEDDISPLAYNAME)</summary>
+        case LocaleString_LocalizedDisplayName:
+            value = [[gbLocale displayNameForKey:NSLocaleIdentifier value:currentLocale.localeIdentifier] UTF8String];
+            break;
+        /// <summary>Display name (language + country usually) in English, eg "German (Germany)" (corresponds to LOCALE_SENGLISHDISPLAYNAME)</summary>
+        case LocaleString_EnglishDisplayName:
+            value = [[gbLocale displayNameForKey:NSLocaleIdentifier value:currentLocale.localeIdentifier] UTF8String];
```

Comment:
If this is the same as `LocaleString_LocalizedDisplayName`, should it read
```
case LocaleString_LocalizedDisplayName:
case LocaleString_EnglishDisplayName:
            value = [[gbLocale displayNameForKey:NSLocaleIdentifier value:currentLocale.localeIdentifier] UTF8String];
            break;
```

Reply:
Updated